src/live/src/user-to-user/live-service.js

```diff
@@ -0,0 +1,335 @@
+    this.__pubnubListener = value;
+  }
+
+  get _pubnubListener() {
+    if (!this.__pubnubListener) {
+      this._throwNotInitializedError();
+    }
+    return this.__pubnubListener;
+  }
+
+  isInitialized() {
+    return !!this.__pubnubClient && !!this.__pubnubListener && !!this._registeredUser;
+  }
+
+  /**
+   * Registers the active user for live service
+   * @returns {Promise}
+   */
+  registerUser(user) {
+    if (!user || !user.isActive()) {
+      return Promise.reject(new ActiveUserError('Missing or invalid active user'));
+    }
+
+    return this._makeRegisterRequest(user._id)
+      .then((regResponse) => {
+        this._registeredUser = user;
+        this._userChannelGroup = regResponse.userChannelGroup;
+        const config = extend({
+          ssl: true,
+          authKey: this._registeredUser._kmd.authtoken
+        }, regResponse);
+        return config;
+      });
+  }
+
+  /**
+   * @param {PubNub} pubnubClient
+   * @param {PubNubListener} pubnubListener
+   */
+  initialize(pubnubClient, pubnubListener) {
+    this._pubnubListener = pubnubListener;
+    this._pubnubClient = pubnubClient;
+    this._pubnubClient.addListener(this._pubnubListener);
+    this._subscribeToUserChannelGroup();
+  }
+
+  /**
+   * Unsubscribes from all events in PubNub client and in listener.
+   * Unregisters user from live service
+   */
+  uninitialize() {
+    this.unsubscribeFromAll();
+    this._pubnubClient = null;
+    this._pubnubListener = null;
+  }
+
+  onConnectionStatusUpdates(func) {
+    this._pubnubListener.on(PubNubListener.unclassifiedEvents, func);
+  }
+
+  offConnectionStatusUpdates(func) {
+    if (func) {
+      this._pubnubListener.removeListener(PubNubListener.unclassifiedEvents, func);
+    } else {
+      this._pubnubListener.removeAllListeners(PubNubListener.unclassifiedEvents);
+    }
+  }
+
+  /**
+   * @param {string} channelName The name of the PubNub channel to publish to
+   * @param {Object} message The message to be published
+   * @returns {Promise}
+   */
+  publishToChannel(channelName, message) {
+    if (!this.isInitialized()) {
+      return Promise.reject(new KinveyError('Live service is not initialized'));
+    }
+
+    const validationErr = this._validatePublishData(channelName, message);
+    if (validationErr) {
+      return Promise.reject(validationErr);
+    }
+
+    if (isObject(message)) {
+      message.senderId = this._registeredUser._id;
+    }
+
+    return this._pubnubClient.publish({
+      message: message,
+      channel: channelName
+    })
+      .catch((err) => {
+        err = err.status.errorData;
+        const resp = new Response({ data: err, statusCode: err.status, headers: err.response.headers });
+        return Promise.reject(resp);
+      });
+  }
+
+  /**
+   * Start listening for events for specified channel
+   * @param {string} channelName
+   * @param {LiveServiceReceiver} receiver
+   */
+  subscribeToChannel(channelName, receiver) {
```

Comment:
No, the answer to this is the same as the above. I should have read ahead. You can refer to the doc (which is once again public lol) on what the public (SDK consumer facing) API is.